android/airship-framework-proxy/src/main/java/com/urbanairship/android/framework/proxy/TagOperation.kt

```diff
@@ -0,0 +1,28 @@
+package com.urbanairship.android.framework.proxy
+
+import com.urbanairship.channel.TagEditor
+import com.urbanairship.channel.TagGroupsEditor
+import com.urbanairship.json.JsonMap
+
+public enum class TagOperationAction {
+    ADD, REMOVE
+}
+
+public data class TagOperation(
+    public val tags: List<String>,
+    public val action: TagOperationAction
+) {
+    public constructor(json: JsonMap) : this(
+        tags = json.require("tags").requireList().map { it.requireString() },
+        action = TagOperationAction.valueOf(
+            json.require("operationType").requireString().uppercase()
+        )
+    )
```

Comment:
how do we handle exceptions? if json is invalid will it crash the app or just print an exception?

Reply:
The proxy methods are only used internally by our frameworks, then our frameworks should have some editors that wrap the values correctly so I think it should be ok